state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -1965,6 +1969,284 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
+			*id = id.saturating_add(1);
+			*id
+		});
+
+		// Only boost on non-zero fee and if the channel isn't already boosted:
+		if T::SafeMode::get().boost_deposits_enabled &&
+			boost_fee > 0 &&
+			!matches!(boost_status, BoostStatus::Boosted { .. })
+		{
+			match Self::try_boosting(asset, amount, boost_fee, prewitnessed_deposit_id) {
+				Ok(BoostOutput { used_pools, total_fee: boost_fee_amount }) => {
+					let amount_after_boost_fee = amount.saturating_sub(boost_fee_amount);
+
+					// Note that ingress fee is deducted at the time of boosting rather than the
+					// time the deposit is finalised (which allows us to perform the channel
+					// action immediately):
+					let AmountAndFeesWithheld { amount_after_fees, fees_withheld: ingress_fee } =
+						Self::conditionally_withhold_ingress_fee(
+							asset,
+							amount_after_boost_fee,
+							&origin,
+						);
+
+					let used_pool_tiers = used_pools.keys().cloned().collect();
+
+					if let Ok(action) = Self::perform_channel_action(
+						action,
+						asset,
+						source_address,
+						amount_after_fees,
+						origin.clone(),
+					) {
+						Self::deposit_event(Event::DepositBoosted {
+							deposit_address,
+							asset,
+							amounts: used_pools,
+							block_height,
+							prewitnessed_deposit_id,
+							channel_id,
+							deposit_details: deposit_details.clone(),
+							ingress_fee,
+							boost_fee: boost_fee_amount,
+							action,
+							origin_type: origin.into(),
+						});
+					} else {
+						// TODO: emit error?
+					}
+
+					return Some(BoostStatus::Boosted {
+						prewitnessed_deposit_id,
+						pools: used_pool_tiers,
+						amount,
+					});
+				},
+				Err(err) => {
+					Self::deposit_event(Event::InsufficientBoostLiquidity {
+						prewitnessed_deposit_id,
+						asset,
+						amount_attempted: amount,
+						channel_id,
+					});
+					log::debug!(
+						"Deposit (id: {prewitnessed_deposit_id}) of {amount:?} {asset:?} and boost fee {boost_fee} could not be boosted: {err:?}"
+					);
+				},
+			}
+		}
+
+		None
+	}
+
+	fn process_vault_swap_request_prewitness(
+		block_height: TargetChainBlockNumber<T, I>,
+		VaultDepositWitness {
+			input_asset: asset,
+			deposit_address,
+			channel_id,
+			deposit_amount: amount,
+			deposit_details,
+			output_asset,
+			destination_address,
+			deposit_metadata,
+			tx_id,
+			broker_fee,
+			affiliate_fees,
+			refund_params,
+			dca_params,
+			boost_fee,
+		}: VaultDepositWitness<T, I>,
+	) {
+		let destination_address_internal =
+			match T::AddressConverter::decode_and_validate_address_for_asset(
+				destination_address.clone(),
+				output_asset,
+			) {
+				Ok(address) => address,
+				Err(err) => {
+					log::warn!("Failed to process vault swap due to invalid destination address. Tx hash: {tx_id:?}. Error: {err:?}");
+					return;
+				},
+			};
+
+		let broker = broker_fee.account.clone();
+
+		let broker_fees = Self::assemble_broker_fees(broker_fee, affiliate_fees);
```

Comment:
I prefer to keep it the way it is (for the same reason as the other as in the other comment). I also wouldn't inline even if we only used it once since it makese sense to keep it separate conceptually and it makes the rest of the code cleaner.

> though wouldn't mind keeping the function and writing a test

Added a test that uses max affiliates